base/abstractarraymath.jl

```diff
@@ -4,6 +4,15 @@
 
 isreal(x::AbstractArray) = all(isreal,x)
 iszero(x::AbstractArray) = all(iszero,x)
+isone(x::AbstractVector) = false # vectors are not multiplicative identities
+function isone(x::AbstractMatrix)
+    m, n = size(x)
+    m != n && return false # only square matrices can satisfy x == one(x)
+    @inbounds for i in 1:m
```

Comment:
FWIW, the implementation that I tried was variations of:
```julia
...
@inbounds for i in 1:m*m
    if (i-1) % (m+1) == 0
        !isone(x[i]) && return false
    else
        !iszero(x[i]) && return false
    end
end
return true
...
```
That was obviously dumb. I have a faster implementation with two loops that I'm pushing now.